crates/bevy_ecs/src/system/exclusive_function_system.rs

```diff
@@ -110,18 +110,12 @@ where
 
     #[inline]
     unsafe fn run_unsafe(
-        &mut self,
-        _input: SystemIn<'_, Self>,
-        _world: UnsafeWorldCell,
-    ) -> Self::Out {
-        panic!("Cannot run exclusive systems with a shared World reference");
-    }
-
-    fn run_without_applying_deferred(
         &mut self,
         input: SystemIn<'_, Self>,
-        world: &mut World,
+        world: UnsafeWorldCell,
     ) -> Self::Out {
+        // SAFETY: `is_exclusive` returned true, so the caller ensured we had `&mut World` access
```

Comment:
This safety comment doesn't make sense to me. We never check is_exclusive in this method. My understanding here is that you've passed the safety requirements for calling `world_mut` to the caller as it can't be verified here. https://docs.rs/bevy/latest/bevy/ecs/world/unsafe_world_cell/struct.UnsafeWorldCell.html#safety-1

And so need to update the safety comment on `run_unsafe`.

Reply:
Right, I did update the safety comment on `run_unsafe`.  It now has [the additional requirement](https://github.com/bevyengine/bevy/pull/18406/files/e9dc5f56843472171c867754e4c8512fe950d0a2#diff-75be02e2bfe4197aaed28e4d0e5dd3e3bfe3c3f464817ea896bae6b3b128425cR72-R73) that 

```rust
    /// - If [`System::is_exclusive`] returns `true`, then it must be valid to call
    ///   [`UnsafeWorldCell::world_mut`] on `world`.
```

What I wanted to communicate here is that because `ExclusiveFunctionSystem::is_exclusive` returns `true`, that clause triggered and the caller had to ensure that `world_mut` was valid.  

If you have ideas on how to make the wording more clear, I'm happy to update it!